homeassistant/components/somfy_mylink/cover.py

```diff
@@ -57,101 +55,57 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class SomfyShade(RestoreEntity, CoverEntity):
     """Object for controlling a Somfy cover."""
 
+    _attr_should_poll = False
+    _attr_assumed_state = True
+
     def __init__(
         self,
         somfy_mylink,
         target_id,
         name="SomfyShade",
         reverse=False,
-        device_class=DEVICE_CLASS_WINDOW,
+        device_class=CoverDeviceClass.WINDOW,
     ):
         """Initialize the cover."""
         self.somfy_mylink = somfy_mylink
-        self._target_id = target_id
-        self._name = name
+        self._attr_unique_id = self._target_id = target_id
```

Comment:
I wasn't 100% sure about whether it was useful to keep both assignements here (`_attr_unique_id` and `_target_id`) but in the end I felt it made the code clearer.
The other option is to replace all the references to `self._target_id` with `self._attr_unique_id` like this `await self.somfy_mylink.move_down(self._attr_unique_id)`, but I felt it added confusion

Reply:
It still can be useful, as _attr_unique_id has been typed `str | None`, while `_target_id` can't be None.

Nevertheless, please put the assignments on separate lines.

Reply:
```suggestion
        self._target_id = target_id
        self._attr_unique_id = target_id
```